components/admin_console/system_users/system_users.jsx

```diff
@@ -169,65 +156,38 @@ export default class SystemUsers extends React.Component {
         } = this.props.actions;
 
         if (this.props.teamId === SearchUserTeamFilter.ALL_USERS) {
-            getProfiles(page + 1, USERS_PER_PAGE, {}).then(this.loadComplete);
+            await getProfiles(page + 1, USERS_PER_PAGE, {});
         } else if (this.props.teamId === SearchUserTeamFilter.NO_TEAM) {
-            loadProfilesWithoutTeam(page + 1, USERS_PER_PAGE).then(this.loadComplete);
+            await loadProfilesWithoutTeam(page + 1, USERS_PER_PAGE);
         } else {
-            const {data} = await loadProfilesAndTeamMembers(page + 1, USERS_PER_PAGE, this.props.teamId);
-            if (data) {
-                this.loadComplete();
-            }
-        }
-    }
-
-    search(term, teamId = this.props.teamId, filter = '') {
-        if (term === '') {
-            this.setState({
-                loading: false,
-            });
-
-            this.searchTimeoutId = '';
-            return;
+            await loadProfilesAndTeamMembers(page + 1, USERS_PER_PAGE, this.props.teamId);
         }
-
-        this.doSearch(teamId, term, filter);
+        this.setState({loading: false});
     }
 
-    doSearch(teamId, term, filter, now = false) {
-        clearTimeout(this.searchTimeoutId);
-
+    doSearch = debounce(async (term, teamId, filter = '') => {
         this.setState({loading: true});
 
-        const options = this.getFilterOptions(filter);
+        const options = {
+            ...this.getFilterOptions(filter),
+            ...teamId && {team_id: teamId},
+            ...teamId === SearchUserTeamFilter.NO_TEAM && {
+                [UserSearchOptions.WITHOUT_TEAM]: true,
+            },
+        };
 
-        if (teamId === SearchUserTeamFilter.NO_TEAM) {
-            options[UserSearchOptions.WITHOUT_TEAM] = true;
+        const {data: profiles} = await this.props.actions.searchProfiles(term, options);
```

Comment:
Actually, debouncing and side-effect here should move into redux middleware (thunk or something) in Redux strategy, not in the component

Reply:
Using the return value of action creator directly in components is not a correct usage of redux.